indico/modules/rb_new/client/js/modules/calendar/reducers.js

```diff
@@ -112,11 +122,85 @@ function acceptPrebooking(calendar, bookingId, roomId) {
 }
 
 export default combineReducers({
-    request: requestReducer(calendarActions.FETCH_REQUEST, calendarActions.FETCH_SUCCESS, calendarActions.FETCH_ERROR),
+    requests: combineReducers({
+        calendar: requestReducer(
+            calendarActions.FETCH_CALENDAR_REQUEST,
+            calendarActions.FETCH_CALENDAR_SUCCESS,
+            calendarActions.FETCH_CALENDAR_ERROR
+        ),
+        activeBookings: requestReducer(
+            calendarActions.FETCH_ACTIVE_BOOKINGS_REQUEST,
+            calendarActions.FETCH_ACTIVE_BOOKINGS_SUCCESS,
+            calendarActions.FETCH_ACTIVE_BOOKINGS_ERROR,
+        )
+    }),
     filters: filterReducerFactory('calendar', initialRoomFilterStateFactory, processRoomFilters),
+    activeBookings: (state = initialActiveBookingsState, action) => {
+        switch (action.type) {
+            case calendarActions.ACTIVE_BOOKINGS_RECEIVED: {
+                const {bookings: newBookings, rowsLeft} = camelizeKeys(action.data);
+                const {data} = state;
+                const newData = {...data};
+
+                Object.entries(newBookings).forEach(([date, bookings]) => {
+                    if (!(date in newData)) {
+                        newData[date] = bookings;
+                    } else {
+                        newData[date].push(...bookings);
```

Comment:
mutating state, the lists are never copied

    newData[date] = [...newData[date], ...bookings]